src/controllers/todo-lists-controller.js

```diff
@@ -32,75 +43,113 @@ router.patch('/:id/availability', (req, res) => {
 })
 
 /**
- * PATCH /lists/:id/items/:index/ready
+ * PATCH /lists/:id/items/:index/ready?status=:status
+ * { "status": true }
  *
- * Marks the item [index] from the list as ready.
+ * Changes the item [index] ready status from the list as [status].
  */
-router.patch('/:id/items/:index/ready', (req, res) => {
+ router.patch('/:id/items/:index/ready', (req, res) => {
     const listId = req.params.id
     const itemIndex = req.params.index
-    TodoListsController.handleResult(listsService.markReadiness(listId, itemIndex, true), res)
+    const ready = req.query.status
+    TodoListsController.handleResult(
+        listsService.performUpdateItemReadyStatus(listId, itemIndex, ready), res
+    )
 })
 
 /**
- * PATCH /lists/:id/items/:index/unready
+ * PATCH /lists/:id/items/:index/position
+ * { "new_index": 4 }
  *
- * Marks the item [index] from the list as unready.
+ * Moves the item [index] to the new index.
  */
-router.patch('/:id/items/:index/unready', (req, res) => {
+router.patch('/:id/items/:index/position', (req, res) => {
     const listId = req.params.id
     const itemIndex = req.params.index
-    TodoListsController.handleResult(listsService.markReadiness(listId, itemIndex, false), res)
+    const newIndex = req.body.new_index
+    TodoListsController.handleResult(
+        listsService.performUpdateItemPosition(listId, itemIndex, newIndex), res
+    )
 })
 
 /**
- * PUT /lists/:id/items/:index
- * { "item": "take a shower" }
+ * DELETE /lists/:id/items/:index
  *
- * Replaces the item [index] with the given.
+ * Deletes the item [index] from the list.
  */
-router.put('/:id/items/:index', (req, res) => {
+ router.delete('/:id/items/:index', (req, res) => {
     const listId = req.params.id
-    const index = req.params.index
-    const item = req.body.item
-    TodoListsController.handleResult(listsService.modifyElement(listId, index, item), res)
+    const itemIndex = req.params.index
+    TodoListsController.handleResult(listsService.performDeleteItem(listId, itemIndex), res)
 })
 
 /**
- * PATCH /lists/:id/items/:index/position
- * { "new_index": 4 }
- *
- * Moves the item [index] to the new index.
+ * POST /lists/commit
+ * { "list" : [list] }
+ * 
+ * Commits the updated list that other instances verified on top of the previous list. 
+ * It also makes the list available after the change is made.
+ * 
  */
-router.patch('/:id/items/:index/position', (req, res) => {
-    const listId = req.params.id
-    const itemIndex = req.params.index
-    const newIndex = req.body.new_index
-    TodoListsController.handleResult(listsService.moveElement(listId, itemIndex, newIndex), res)
-})
+ router.post('/commit', (req, res) => {
+    const newList = req.body.list
+
+    const list = listsService.createList(newList)
+
+    if (list) {
+        console.log("Successful commit: list created")
+        return res.status(200).json({ list: list })
```

Comment:
```suggestion
        res.json({ list: list })
```

Acordate que el return del res no se necesita en ningún lado